crates/service-client/src/lambda.rs

```diff
@@ -305,13 +314,8 @@ impl LambdaError {
     /// retrying can succeed.
     pub fn is_retryable(&self) -> bool {
         match self {
-            LambdaError::Body(err) => err.is_retryable(),
```

Comment:
`is_retryable` here needs some overhaul. First, this method here is used only within the service discovery retry code, so doesn't affect correctness of invocations (and it shouldn't be, because detecting whether an error is retryable or not can be very fragile).

In the specific case of `LambdaError::Body`, this variant is constructed when reading the body that comes from either invoker/service discovery. For the case of discovery, this error never exists, as we construct the body with `http_body_util::Full` (which can never fail), while for the invoker it can fail only if the input channel is closed, but in that case it's hard to say whether is retryable or not. And in any case, right now this method is not used in the invoker.